nullaway/src/test/java/com/uber/nullaway/LegacyVarargsTests.java

```diff
@@ -200,7 +200,8 @@ public void testSkipJetbrainsNotNullOnVarArgsFromThirdPartyJars() {
             "    ThirdParty.takesNullableVarargs(o1);", // Empty var args passed
             "    ThirdParty.takesNullableVarargs(o1, o4);",
             "    ThirdParty.takesNullableVarargs(o1, (Object) null);",
-            "    // BUG: Diagnostic contains: passing @Nullable parameter '(Object[]) null' where @NonNull",
+            "    // No error: we require a type-use annotation for nullability of the varargs array itself",
+            // TODO should we preserve the old behavior in legacy mode?
```

Comment:
@lazaroclapp with this change, we no longer detect an error when passing a `null` array in the varargs position to a third-party method with a JetBrains `@NotNull` annotation on the varargs parameter.  On the one hand, getting an error here is the "desired" behavior, in that the annotation is supposed to apply to the array itself (see #720).  But, everywhere else, a declaration annotation on the varargs argument is treated as applying to the contents of the array, not the array itself.  To me, adding the extra code complexity to detect these errors in legacy mode for this corner case is not worth it; hopefully Kotlinc will be convinced to put annotations in the "right" place soon.  But I'd be interested in your opinion.

Reply:
I think without a coordinated change on `kotlinc`, I would prefer to keep the hack. But I don't feel super strongly about it. It's just an additional test of (a) is the annotation in bytecode, (b) is it the jetbrains annotation, (c) is it in var args... if all that is true, hacky override with a comment linking to the issue. No? Would this be hard to maintain?

Unfortunately, Java code interfacing with Kotlin code is a big portion of what we will see in Android for the foreseeable future, so `kotlinc` quirks might be the kind of thing we can't ignore for the sake of uniformity. Though if you can convince them to generate the "right" bytecode, I'd be happy to see this hack disappear (hopefully we don't need then to analyze magic numbers to see which `kotlinc` version produced the jar)